### ISSUE TITLE
fieldObserver: remove or fix aria-disabled logic

### ISSUE THREAD

**Issue by oddvernes** [digdir/designsystemet#3367]
Inside `fieldObserver`, there is logic to add `aria-disabled` to `label` etc if `input` has disabled/aria-disabled. This was done due to axe tests failing, but are not real requirements in the wai-aria 1.2 rules. Curently disabled and aria-disabled are not observed by the mutation observer, therefore, if an input is disabled at load, the label will have the aria-`disabled="true"` attribute even after input has been enabled programmatically. 

Either disabled/aria-disabled should be "observed", or the logic removed (which will make axe complain). @eirikbacker suggests we remove the logic.

**Comment by unekinn** [comment]
Which Axe rule is triggering the false positive? Is there an issue in axe-core about this already? If not, we should submit a bug report.

If it is possible to locally disable the Axe rule without leaving the door open for actual violations, that would be ideal.

**Comment by oddvernes** [comment]
This was added months ago and I wasn't involved, but I'm guessing it is this, the color contrast check is triggered unless the elements are aria-disabled

<img width="1231" alt="Image" src="https://github.com/user-attachments/assets/89b722ae-bae2-41a3-a774-119433fe9885" />

So it wasn't a direct aria-rule per se. We can locally disable the contrast check with this in the storybook parametres
```
a11y: {
      config: {
        rules: [{ id: 'color-contrast', selector: ':disabled' }],
      },
    },

**Comment by oddvernes** [comment]
I forget, but I probably went in and added disabled to the story specifically to trigger the violation above 😅. In which case we can just remove the logic and call it a day 👍
